File: ecmascript/pangram/pangram.js
class Pangram {
  constructor (phrase) {
    this.phrase = phrase
  }
  
  isPangram () {
    const alphabet = 'abcdefghijklmnopqrstuvwxyz'
    
    const result = this.phrase
      .toLowerCase()
      .split('')
      .filter(char => new Boolean(char.trim()))
      .reduce((chars, char) => {
        if (chars.includes(char)) {
          const new_chars_array = chars.split('')
          new_chars_array.splice(chars.indexOf(char), 1)
          return new_chars_array.join('')
        }
        
        return chars
      }, alphabet)

    return !result.length
  }
}

export default Pangram

File: python/pangram/pangram.py
def is_pangram(phrase):
    alphabet = ''.join([chr(i) for i in range(97, 122 + 1)])

    for char in list(phrase.lower()):
        if char in alphabet:
            parts = alphabet.partition(char)
            alphabet = parts[0] + parts[2]

    return not len(alphabet)
